app/misc/git.ts

```diff
@@ -419,6 +422,35 @@ function pushToRemote() {
     });
 }
 
+//Takes the number of local commits and the number of remote commits and returns the difference
+//This will be the total number of unpushed commits
+@Injectable()
+export function calcUnpushedCommits() {
+  var calc = 0;
+
+  for (let i = 0 ; i < 5; i++){
```

Comment:
I'm a bit lost with this for loop. Why are we doing it 5 times?

Reply:
We made it 5-times just to make sure that the variable is check and updated properly before we use it. It could be twice or one, It was an implementation decision because of the issues we had with calculating the number of unpushed commits locally and the number of commits .